src/condor_dagman/dagman_submit.cpp

```diff
@@ -112,57 +112,57 @@ static std::vector<NodeVar> init_vars(const Dagman& dm, const Job& node) {
 		batchId = dm.options[deep::str::BatchId];
 	}
 
-	vars.push_back(NodeVar("NODE_NAME", nodeName, false));
-	vars.push_back(NodeVar("JOB", nodeName, false));
-	vars.push_back(NodeVar("RETRY", std::to_string(retry), false));
+	vars.emplace_back(NodeVar("NODE_NAME", nodeName, false));
```

Comment:
I think you want

`vars.emplace_back("NODE_NAME", nodeName, false);` 

here -- no `NodeVar` constructor needed.